schemachanger/scplan: add deprules to eval zone cfgs in seqNum order

annrpom opened the pull request:
This patch adds deprules to ensure that zone configs are depended by
each other in increasing`seqNum` order.

Epic: none
Fixes: https://github.com/cockroachdb/cockroach/issues/129482

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/143750)
<!-- Reviewable:end -->

fqazi reviewed (changes requested):
This looks good, just one change that might be unintentional?

> Reviewed 5 of 5 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/143750#-:-OMm1kF11d_jMTsJgCxb:b-za2vwo)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @annrpom)

___
*[`pkg/sql/schemachanger/scplan/internal/rules/registry.go` line 31 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/143750#-OMm1_gHDkEegqICowZJ:-OMm1_gHDkEegqICowZK:b-azshq4) ([raw file](https://github.com/cockroachdb/cockroach/blob/e0391c89c7435142c77554933712d8afbf68b9f5/pkg/sql/schemachanger/scplan/internal/rules/registry.go#L31)):*
> ```Go
> func (r *Registry) ApplyDepRules(ctx context.Context, g *scgraph.Graph) error {
> 	// If expensive logging is enabled, we'll collect stats on the query and
> 	// report on how each dep rule performed.
> ```

Is this an intentional change?


<!-- Sent from Reviewable.io -->

annrpom reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/143750)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @fqazi)

___
*[`pkg/sql/schemachanger/scplan/internal/rules/registry.go` line 31 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/143750#-OMm1_gHDkEegqICowZJ:-OMm2-tL7Z-yushC3GBh:bcrtl4t) ([raw file](https://github.com/cockroachdb/cockroach/blob/e0391c89c7435142c77554933712d8afbf68b9f5/pkg/sql/schemachanger/scplan/internal/rules/registry.go#L31)):*
<details><summary><i>Previously, fqazi (Faizan Qazi) wrote…</i></summary><blockquote>

Is this an intentional change?
</blockquote></details>

oh shoot no ty (i was gonna use it to debug)


<!-- Sent from Reviewable.io -->

annrpom committed:
schemachanger/scplan: add deprules to eval zone cfgs in seqNum order

This patch adds deprules to ensure that zone configs are depended by
each other in increasing`seqNum` order.

Epic: none
Fixes: #129482

Release note: None

fqazi reviewed (approved):
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/143750)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @annrpom)



<!-- Sent from Reviewable.io -->
